modules/echo-server/charts/echo-server/templates/ingress.yaml

```diff
@@ -1,55 +1,55 @@
-{{- if .Values.ingress.enabled -}}
----
+{{- if or (eq (printf "%v" .Values.ingress.nginx.enabled) "true") (eq (printf "%v" .Values.ingress.alb.enabled) "true") -}}
+  {{- $fullName := include "echo-server.fullname" . -}}
+  {{- $svcPort := .Values.service.port -}}
+  {{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1
-kind: IngressClass
-metadata:
-  name: nginx
-spec:
-  controller: k8s.io/ingress-nginx
----
-apiVersion: networking.k8s.io/v1
-{{- $fullName := include "echo-server.fullname" . -}}
-{{- $svcPort := .Values.service.port -}}
-{{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion)) }}
-  {{- if not (hasKey .Values.ingress.annotations "kubernetes.io/ingress.class") }}
-    {{- $_ := set .Values.ingress.annotations "kubernetes.io/ingress.class" .Values.ingress.className}}
-  {{- end }}
-{{- end }}
-## Not yet supported by Forecastle, introduced in Kubernetes 1.19: apiVersion: networking.k8s.io/v1
-## See https://github.com/stakater/Forecastle/issues/128
-{{- if semverCompare ">=1.19-0" .Capabilities.KubeVersion.GitVersion -}}
-apiVersion: networking.k8s.io/v1
-{{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
+  {{- else if semverCompare ">=1.14-0" .Capabilities.KubeVersion.GitVersion -}}
 apiVersion: networking.k8s.io/v1beta1
-{{- else -}}
+  {{- else -}}
 apiVersion: extensions/v1beta1
-{{- end }}
+  {{- end }}
 kind: Ingress
 metadata:
   name: {{ $fullName }}
   annotations:
-    kubernetes.io/ingress.class: {{ .Values.ingress.class }}
-    {{- if (index .Values.ingress "tls_certificate_cluster_issuer") }}
-    cert-manager.io/cluster-issuer: {{ .Values.ingress.tls_certificate_cluster_issuer }}
+    {{- if eq (printf "%v" .Values.ingress.nginx.enabled) "true" }}
+    kubernetes.io/ingress.class: {{ .Values.ingress.nginx.class }}
+      {{- if (index .Values.ingress.nginx "tls_certificate_cluster_issuer") }}
+    cert-manager.io/cluster-issuer: {{ .Values.ingress.nginx.tls_certificate_cluster_issuer }}
+      {{- end }}
+    {{- else if eq (printf "%v" .Values.ingress.alb.enabled) "true" }}
+    kubernetes.io/ingress.class: {{ .Values.ingress.alb.class }}
+    alb.ingress.kubernetes.io/load-balancer-name: {{ index .Values.ingress.alb "load_balancer_name" | default "k8s-common" }}
+    alb.ingress.kubernetes.io/group.name: {{ index .Values.ingress.alb "group_name" | default "common" }}
+    alb.ingress.kubernetes.io/scheme: internet-facing
+    alb.ingress.kubernetes.io/load-balancer-attributes: access_logs.s3.enabled=false
```

Comment:
This kind of annotations need to be moved to a "default" alb created within the alb controller component. After that removed from here. They are not allowed to be changed per ingress, they are global to an ALB (global among users of a single alb.ingress.kubernetes.io/load-balancer-name).

Reply:
besides this comment, everything else looks good

Reply:
@max-lobur @aknysh We do not have a standard mechanism for parallel configuration of Ingresses in that way, and are not about to develop one today for the open source component. Can the `load-balancer-attributes` annotation be safely left off? If not, I say we stick with this and require `echo-server` to use its own Load Balancer if needed.

Reply:
Agreed. Securing annotations should be a separate PR